db/migrate/20220303232114_add_allowed_to_discount_requests.rb

```diff
@@ -0,0 +1,5 @@
+class AddAllowedToDiscountRequests < ActiveRecord::Migration[6.0]
+  def change
+    add_column :discount_requests, :allowed, :boolean, null: false, default: false
```

Comment:
Never thought about that before but I'm wondering if we need `null: false` when default is set.

Reply:
I believe this is good practice once even with the default value, it's still possible to pass `nil` for this field. When `activerecord` does the casting of `nil` it doesn't become a `false`

![image](https://user-images.githubusercontent.com/47258878/156844540-fc0d84d6-af3e-467b-af86-ece0c7046f77.png)

Now I'm thinking about adding some validation for this field as well 🤔

Reply:
I've found [this article](https://medium.com/@mikeappell/rails-presence-validations-booleans-and-pain-1b04adb111ab#:~:text=If%20you%20want%20to%20validate,handles%20boolean%20values%3A%20false.) that explains better the usage for the `presence` validator for a boolean field:

Reply:
We are using [shoulda matchers] (https://matchers.shoulda.io/docs/v5.1.0/) and after adding this validation, this warning started to show up. It makes sense, once it's a boolean field and every value passed will be cast into boolean (for example 1 => true, 0 => false), but I prefer to keep this one, WDYT @brunohkbx ?

![image](https://user-images.githubusercontent.com/47258878/156848907-1b57d9e8-008d-4542-8b25-1e131de08f30.png)

Reply:
Which test?

Reply:
This one: https://github.com/comarev/comarev/pull/78/files#diff-11b99a2daf581c7c0292571d6147e22cfb78c6b6ed9b244b4a90fc5316d248e9R11